### PR TITLE
Core Section 3 Testability Requirements

### PR DESCRIPTION
**Pull request by markymark-26**

### CODE REVIEW

File: core/baseline-core-v1.0-psd01.md [hunk 1]
```diff
@@ -819,65 +825,100 @@ The identifiers and identity utilized in a BPI MUST be controlled by its Princip
...
+3. Verifying if the latest set of keys are in the longest chain using a block explorer such as [Etherscan](https://etherscan.io/).
+
 *Requirements of Identifier registration in DPKI are handled differently from DNS. Although registrars may exist in DPKI, these registrars must adhere to several requirements that ensure that identities belong to the entities they represent. This is achieved the following way:*
 
 #### **[CR3]>[D11]**	
 Private keys utilized in a BPI MUST be generated in a manner that ensures they remain under the Principal Owner’s control. 
 
+[[CR3]>[D11]](#cr3>d11) Testability: To ensure the Principal Owner is solely reponsible for the creation and use the set of public/private keys associated with their DID, a non-custodial wallet must be used to initiate all transactions on behalf of that principal owner within a BPI. An example of implementing this would be to integrate this non-custodial wallet with the BPI APIs that ensure all API payloads/messages are signed, and therefore authorized, solely by the Principal Owner's private key before executing BPI transactions.
```
Comment (chaals) [inline core/baseline-core-v1.0-psd01.md:852]: the fact that you use a non-custodial wallet doesn't seem like it proves the keys are under your control, it's just a best practice. If you tweeted your private key, anyone could use it to sign stuff however they like.

I'm not sure you can enforce this through the key generation. There are standards for key custody that might be good references, e.g. https://csrc.nist.gov/Projects/Key-Management/Key-Management-Guidelines, https://cryptoconsortium.org/cryptocurrency-security-standard-documentation/details/ ...

And a nits - you don't need control of your public key, by definition...

```suggestion
[[CR3]>[D11]](#cr3>d11) Testability: To ensure the Principal Owner is solely reponsible for the creation and use of the private key associated with their DID, a non-custodial wallet must be used to initiate all transactions on behalf of that principal owner within a BPI. An example of implementing this would be to integrate this non-custodial wallet with the BPI APIs that ensure all API payloads/messages are signed, and therefore authorized, solely by the Principal Owner's private key before executing BPI transactions.
```

File: core/baseline-core-v1.0-psd01.md [hunk 2]
```diff
@@ -819,65 +825,100 @@ The identifiers and identity utilized in a BPI MUST be controlled by its Princip
...
+
 #### **[CR4]>[D11]** 
 Generating key pairs in a BPI on behalf of the Principal Owner MUST NOT be allowed.
```
Comment (chaals) [inline core/baseline-core-v1.0-psd01.md:855]: How do you stop me asking a friend to walk through the process? Or asking my head of security to do it for me? And how do you test for that?

File: core/baseline-core-v1.0-psd01.md [hunk 3]
```diff
@@ -819,65 +825,100 @@ The identifiers and identity utilized in a BPI MUST be controlled by its Princip
...
 Principal Owners in a BPI MUST always be in control of their identifiers and the corresponding public keys. 
 
+[[CR5]>[D11]](#cr5>d11) Testability: To guarantee Principal Owners are the sole controller over their identifiers and related public keys, validation can be conducted in the same manner as [[CR2]>[D11]](#cr2>d11). 
+
 #### **[O1]**	
 Principal Owners MAY extend control of their identifier to third parties.
 
 *For example for recovery purposes.*
 
+[[O1]](#o1) Testability: An implementer can verify whether a principal owner has delegated authority of their identifier to third parties by checking in the DID document if there are other DIDs in the [controller](https://www.w3.org/TR/did-core/#did-controller) section of the did document and what type of control these identifiers have by checking the [verification relationships](https://www.w3.org/TR/did-core/#verification-relationships) such as authentication or signing on behalf of [assertion](https://www.w3.org/TR/did-core/#assertion) in the DID document.
```
Comment (chaals) [inline core/baseline-core-v1.0-psd01.md:869]: I don't think you can test for this, except as a process. I can do as outlined, but I can also just write the info in a christmas card for my Mum and tell her to use it wisely.

File: core/baseline-core-v1.0-psd01.md [hunk 4]
```diff
@@ -819,65 +825,100 @@ The identifiers and identity utilized in a BPI MUST be controlled by its Princip
...
+
 #### **[CR9]<[D11]**	
 The rules for registering and renewing identifiers in a DPKI utilized by a BPI MUST be transparent and expressed in simple terms.
 
+[[CR9]>[D11]](#cr9>d11) Testability: The DID method used in a BPI is registered in the DID registry and conformant to the [W3C DID Standard](https://www.w3.org/TR/did-core/).
+
 #### **[R36]**	
 If registration is used as security to an expiration policy, the Principal Owner MUST be explicitly and timely warned that failure to renew the registration on time could result in the Principal Owner losing control of the identifier.
 
+[[R36]](#r36) Testability: The DID method used within a BPI does not have an expiration policy.
```
Comment (chaals) [inline core/baseline-core-v1.0-psd01.md:901]: this doesn't test the requirement except in the edge case of an non-expiring DID (which is probably itself not a good idea in most cases)

File: core/baseline-core-v1.0-psd01.md [hunk 5]
```diff
@@ -819,65 +825,100 @@ The identifiers and identity utilized in a BPI MUST be controlled by its Princip
...
 Once set, namespace rules within a DPKI utilized by a BPI MUST NOT be altered to introduce any new restrictions for renewing or updating identifiers.
 
 *Otherwise, it would be possible to take control of identifiers away from Principal Owners without their consent.*
 
+[[CR10]>[D11]](#cr10>d11) Testability: The DID method used within a BPI has a specification where its rules around the mechanics of a DID method have never been altered. 
+
 #### **[CR11]>[D11]**	
 Within a DPKI utilized by a BPI, processes for renewing or updating identifiers MUST NOT be modified to introduce new restrictions for updating or renewing an identifier, once issued.
 
+[[CR11]>[D11]](#cr11>d11) Testability: The DID method used within a BPI has a specification for which the read, create, update, and delete DID operations have never been altered.
```
Comment (chaals) [inline core/baseline-core-v1.0-psd01.md:913]: I'm not sure the requirement here is a good idea. What if the processes were insufficiently secure, and the change is one that makes them secure, and works?

**Review by chaals — CHANGES_REQUESTED** [review]
I got halfway through on this pass, more work to do. But so far I think there are some things that need changes. (Although I might be wrong about them - this is half of a first-pass review).
